packages/contracts-bedrock/contracts/test/Bytes.t.sol

```diff
@@ -0,0 +1,173 @@
+    /// @dev Tests that the `slice` function works as expected when starting from
+    /// index 0.
+    function test_slice_fromZeroIdx_works() public {
+        bytes memory input = hex"11223344556677889900";
+
+        // Exhaustively check if all possible slices starting from index 0 are correct.
+        assertEq(Bytes.slice(input, 0, 0), hex"");
+        assertEq(Bytes.slice(input, 0, 1), hex"11");
+        assertEq(Bytes.slice(input, 0, 2), hex"1122");
+        assertEq(Bytes.slice(input, 0, 3), hex"112233");
+        assertEq(Bytes.slice(input, 0, 4), hex"11223344");
+        assertEq(Bytes.slice(input, 0, 5), hex"1122334455");
+        assertEq(Bytes.slice(input, 0, 6), hex"112233445566");
+        assertEq(Bytes.slice(input, 0, 7), hex"11223344556677");
+        assertEq(Bytes.slice(input, 0, 8), hex"1122334455667788");
+        assertEq(Bytes.slice(input, 0, 9), hex"112233445566778899");
+        assertEq(Bytes.slice(input, 0, 10), hex"11223344556677889900");
+    }
+
+    /// @dev Tests that the `slice` function works as expected when starting from
+    /// indexes [1, 9] with lengths [1, 9], in reverse order.
+    function test_slice_fromNonZeroIdx_works() public {
+        bytes memory input = hex"11223344556677889900";
+
+        // Exhaustively check correctness of slices starting from indexes [1, 9]
+        // and spanning [1, 9] bytes, in reverse order
+        assertEq(Bytes.slice(input, 9, 1), hex"00");
+        assertEq(Bytes.slice(input, 8, 2), hex"9900");
+        assertEq(Bytes.slice(input, 7, 3), hex"889900");
+        assertEq(Bytes.slice(input, 6, 4), hex"77889900");
+        assertEq(Bytes.slice(input, 5, 5), hex"6677889900");
+        assertEq(Bytes.slice(input, 4, 6), hex"556677889900");
+        assertEq(Bytes.slice(input, 3, 7), hex"44556677889900");
+        assertEq(Bytes.slice(input, 2, 8), hex"3344556677889900");
+        assertEq(Bytes.slice(input, 1, 9), hex"223344556677889900");
+    }
+
+    /// @dev Tests that the `slice` function works as expected when slicing between
+    /// multiple words in memory. In this case, we test that a 2 byte slice between
+    /// the 32nd byte of the first word and the 1st byte of the second word is
+    /// correct.
+    function test_slice_acrossWords_works() public {
+        bytes
+            memory input = hex"00000000000000000000000000000000000000000000000000000000000000112200000000000000000000000000000000000000000000000000000000000000";
+
+        assertEq(Bytes.slice(input, 31, 2), hex"1122");
+    }
+
+    /// @dev Tests that, when given an input bytes array of length `n`,
+    /// the `slice` function will always revert if `_start + _length > n`.
+    function testFuzz_slice_outOfBounds_reverts(
+        bytes memory _input,
+        uint256 _start,
+        uint256 _length
+    ) public {
+        // We want a valid start index and a length that will not overflow.
+        vm.assume(_start < _input.length && _length < type(uint256).max - 31);
```

Comment:
could we use `bound()` here instead to increase the number of test cases that actually run?

Reply:
Same for other locations using vm.assume()

Reply:
Will do 👍